File: android/app/src/main/java/com/clingsync/android/presentation/WorkUpdate.kt
package com.clingsync.android.presentation

import com.clingsync.android.FileStatus
import java.util.UUID

// A normalized WorkManager signal. The ViewModel produces it from the raw
// List<WorkInfo> plus the parsed status/result JSON, so the reducer never sees
// androidx.work types and never touches the filesystem.
sealed interface WorkUpdate {
    data class Enqueued(val id: UUID) : WorkUpdate

    data class Running(
        val statuses: Map<String, FileStatus>,
        val uploadedBytes: Long,
    ) : WorkUpdate

    data class Succeeded(val finalStatuses: Map<String, FileStatus>) : WorkUpdate

    data class Failed(val error: String) : WorkUpdate

    data object Cancelled : WorkUpdate
}

File: ios/ClingSync/WorkUpdate.swift
import Foundation

// A normalized upload signal. The store produces it from the upload mechanism's
// persisted progress/result, so the reducer never sees the transport types and
// never touches the filesystem.
enum WorkUpdate {
    case enqueued(id: UUID)
    case running(statuses: [String: FileStatus], uploadedBytes: Int64)
    case succeeded(finalStatuses: [String: FileStatus])
    case failed(error: String)
    case cancelled
}
